pkg/uroot/uroot.go

```diff
@@ -62,8 +52,8 @@ func DefaultRamfs() *cpio.Archive {
 			cpio.CharDev("dev/null", 0666, 1, 3),
 			cpio.CharDev("dev/port", 0640, 1, 4),
 			cpio.CharDev("dev/urandom", 0666, 1, 9),
-			cpio.StaticFile("etc/resolv.conf", nameserver, 0644),
```

Comment:
could you just mount a tmpfs on /etc? You're changing /etc anyway.

Reply:
Mounting tmpfs on etc breaks this usage. 
```
u-root -files "my/resolv.conf:etc/resolv.conf"
```

Mounting tmpfs on etc will overwrite all static files put by users. To be backward compatible, we must keep the old etc files. In my implementation, a user can still put static configured files for their needs. The init process only creates files on tmpfs. For example, a static DNS server is required or a fixed timezone info is required. 

The static resolv.conf file does not work in uroot's dhclient implementation though. dhclient overwrites the static file anyway. However, it's by design of dhclient. We may discuss that topic if needed in a different thread but for now this PR is sufficient to fix everything else. The dhclient API does not need to change because overwriting is the correct implementation. The GNU implementation uses hook scripts to allow user rewrite the behaviors.

Talking about the user scenarios, I feel the only needs for a static DNS server IP is when the IP address is static as well. On the other hand, it's not affected in the case that the DHCP server does not provide the DNS server. A static configured resolv.conf will be used in this case.

Secondly, not all etc files are suitable to be placed on tmpfs. Some of them should not be modified because the processes will not reload the new settings.

Reply:
@MedicineYeh is right for read-only filesystems like squashfs.

Reply:
bleah. we really wanted 0 /etc files in u-root originally, and this sort of thing is the reason why. We could actually use layering mounts if they were not so terrible ....

wow this is annoying. :-)

Reply:
Yeah. etc file is just something never die. It's the best design if you don't want a complicated solution. There are many troubles if we really want to use DNS forwarder.

What is the review status? I don't see any action items or open questions not answered.

Reply:
As discussed in slack. I made another version which copies the etc files and mount back on to etc.